applications/MeshingApplication/custom_processes/metrics_hessian_process.cpp

```diff
@@ -107,33 +107,34 @@ array_1d<double, 3 * (TDim - 1)> ComputeHessianSolMetricProcess::ComputeHessianM
     double interpolation_error = mInterpError;
     if (mEstimateInterpError) {
-        interpolation_error = mesh_constant * MathUtils<double>::Max(ElementMaxSize, ElementMaxSize * norm_frobenius(hessian_matrix));
+        interpolation_error = mMeshConstant * MathUtils<double>::Max(NodalH, NodalH * norm_frobenius(hessian_matrix)); // NOTE: To compute it properly instead of iterating over the nodes you should iterate over the elements and instead of ElementMaxSize you should iterate over the edges, this is equivalent when using nodes and computing NodalH previously
     }
 
-    // We check is the interpolation error is near zero. If it is we will correct it
-    if (interpolation_error < std::numeric_limits<double>::epsilon()) {
-        KRATOS_WARNING("ComputeHessianSolMetricProcess") << "WARNING: Your interpolation error is near zero: " << interpolation_error  <<  ". Computing a local L(inf) upper bound of the interpolation error"<< std::endl;
-        const double l_square = std::pow(NodalH, 2);
-        for (IndexType i = 0; i < 3 * (TDim - 1); ++i) {
-            interpolation_error = mesh_constant * MathUtils<double>::Max(interpolation_error, l_square * std::abs(rHessian[i]));
-        }
-    }
-    const double c_epslilon = mMeshConstant/interpolation_error;
-    const double min_ratio = 1.0/(ElementMinSize * ElementMinSize);
-    const double max_ratio = 1.0/(ElementMaxSize * ElementMaxSize);
-
     // Declaring the eigen system
     MatrixType eigen_vector_matrix, eigen_values_matrix;
 
     MathUtils<double>::EigenSystem<TDim>(hessian_matrix, eigen_vector_matrix, eigen_values_matrix, 1e-18, 20);
 
-    // Recalculate the Metric eigen values
-    for (IndexType i = 0; i < TDim; ++i)
-        eigen_values_matrix(i, i) = MathUtils<double>::Min(MathUtils<double>::Max(c_epslilon * std::abs(eigen_values_matrix(i, i)), max_ratio), min_ratio);
+    // We check is the interpolation error is near zero. If it is we will correct it
+    if (interpolation_error < std::numeric_limits<double>::epsilon()) { // In practice, the Hessian of function u can be 0, e.g. if u is linear, then |Hu| is not definite. In this particular case, the interpolation error is 0 and we want to prescribe a mesh size which is infinite. To solve this issue, this infinite size prescription is truncated by imposing lambda maximal size hmax . This is equivalent to truncate tiny eigenvalues by lambda  = 1/hmax^2 . This is taken from page 34 on Metric-Based Anisotropic Mesh Adaptation
+        KRATOS_WARNING("ComputeHessianSolMetricProcess") << "WARNING: Your interpolation error is near zero: " << interpolation_error  <<  ". Computing a local L(inf) upper bound of the interpolation error"<< std::endl;
+        const double l_square_minus1 = 1.0/std::pow(ElementMaxSize, 2);
+        for (IndexType i = 0; i < TDim; ++i) {
+            eigen_values_matrix(i, i) = l_square_minus1;
```

Comment:
I would be more consistent:
```
eigen_values_matrix(i, i) = MathUtils<double>::Max(l_square_minus1,std::abs(eigen_values_matrix(i, i)) * mMeshConstant/interpolation_error));
```
We already have eigen_values_matrix no?

Reply:
If interpolation_error is almost 0 mMeshConstant/interpolation_error -> Infinity, then is better this way

Reply:
To avoid numerical issues